File: algorithms/lab5/linkedList.js
function prepend(x, xs) {
    return function (select) {
        return select(x, xs)
    }
}
function select_head(x, xs) { return x }
function select_tail(x, xs) { return xs }

function head(a) { return a(select_head) }
function tail(a) { return a(select_tail) }
function nil() { return nil }

var linked_list = prepend('1', prepend('3', prepend('7', nil)));
// 'a' -> 'b' -> nil

head(linked_list); // => 'a'
head(tail(linked_list)); // => 'b'
head(tail(tail(linked_list))); // => nil

var arr = [];

while (linked_list !== nil) {
    console.log(head(linked_list));
    arr[arr.length] = head(linked_list);
    linked_list = tail(linked_list)
}

// console.log(arr);

for (i = 0; i < arr.length; i++) {
    if (arr[i] < arr[i - 1]) {
        var result = false
    } else { result = true }
}

console.log('Is sorted? ', result)

File: algorithms/lab5/linked_list.py
class Node:

    def __init__(self):
        self.data = None
        self.next = None

    def __str__(self):
        return "Data: %s, Next -> %s" % (self.data, self.next)


class LinkedList:
    def __init__(self):
        self.head = Node()
        self.curNode = self.head

    def insert_node(self, data):
        node = Node()
        node.data = data
        node.next = None

        if not self.head.data:
            self.head = node
            self.curNode = node
        else:
            self.curNode.next = node
            self.curNode = node

    def print(self):
        print(self.head)


if __name__ == "__main__":

    l = LinkedList()
    elems = []
    result = None
    number_of_elems = int(input('Введите количество элементов списка: '))

    for i in range(number_of_elems):
        elem = int(input('Введите элемент: '))
        l.insert_node(elem)
        elems.append(elem)

    l.print()

    for i in range(len(elems)):
        if elems[i] < elems[i -1]:
            result = False
        else:
            result = True

    print('Связный список отсортирован?\n', result)
